kotlin/workflow-testing/src/main/java/com/squareup/workflow/testing/RealRenderTester.kt

```diff
@@ -187,7 +187,7 @@ internal class RealRenderTester<PropsT, StateT, OutputT : Any, RenderingT>(
     }
   }
 
-  private inline fun <reified T : Expectation<*>> consumeExpectation(
+  private inline fun <reified T : ExpectedWorkflow<*, *>> consumeExpectedChildWorkflow(
```

Comment:
Both these functions share almost all their logic. The only difference is what happens in the 0-expectations-matched case. This could have been kept as a single function with a function parameter for that behavior, and would make it a lot more clear that they are essentially the same function.

Reply:
I was playing around with using kotlinreflect's typeOf method but having a function parameter makes more sense now that you mention it. I think the other detail is that the function originally returned nonnullable but now would it return null? Is that awkward for `renderChild` and `runningWorker`?  would we uitilize a `requireNotNull` in `renderChild`?